src/model/interactions/application_command.rs

```diff
@@ -652,30 +626,22 @@ impl<'de> Deserialize<'de> for ApplicationCommandInteractionDataOption {
             .and_then(String::deserialize)
             .map_err(DeError::custom)?;
 
-        let value = match map.contains_key("value") {
-            true => Some(
-                map.remove("value")
-                    .ok_or_else(|| DeError::custom("expected value"))
-                    .and_then(Value::deserialize)
-                    .map_err(DeError::custom)?,
-            ),
-            false => None,
-        };
+        // TODO: Does deserializing a `Value` with `Value::deserialize` make sense?
+        let value =
+            map.remove("value").map(Value::deserialize).transpose().map_err(DeError::custom)?;
```

Comment:
@HarmoGlace wouldn't it be enough to just call `map.remove("value")`
```suggestion
        let value = map.remove("value");
```

Reply:
Yeah indeed, I do not remember why I wrote this